boto3/session.py

```diff
@@ -306,11 +308,40 @@ def resource(self, service_name, region_name=None, api_version=None,
 
         :return: Subclass of :py:class:`~boto3.resources.base.ServiceResource`
         """
+        try:
+            resource_model = self._loader.load_service_model(
+                service_name, 'resources-1', api_version)
+        except UnknownServiceError as e:
```

Comment:
That's an interesting point.  I'm thinking yes we need to.

It's also probably worth creating a base Boto3 error that all the boto3 exceptions subclass from, similar to what we do Botocore.